features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/internal/MiscUtils.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.opentelemetry.internal
+
+import android.os.Build
+import com.datadog.android.api.InternalLogger
+
+internal const val NEEDS_DESUGARING_ERROR_MESSAGE =
+    "Trying to use OpenTelemetry SDK support for Android 23 and below. " +
+        "In order for this to properly work you will need to enable coreDesugaring " +
+        "in your compileOptions"
+
+internal fun <T : Any?> executeIfJavaFunctionPackageExists(
+    internalLogger: InternalLogger? = null,
+    defaultActionReturnValue: T,
+    action: () -> T
+): T {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        return action()
+    } else {
+        return try {
+            // we are forcing here the checkup as it could be triggered only at runtime and will be too late
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            Class.forName("java.util.function.Function")
```

Comment:
btw, `Class.forName` may throw not only exceptions, but also Errors https://developer.android.com/reference/java/lang/Class#forName(java.lang.String), they won't be called by catch block in the current state.

Reply:
yup maybe I will change it for `Throwable` instead to make sure.